### PR TITLE
Adding test coverage for PSDrive provider

### PR DESCRIPTION
**Pull request by ChrisUbben**
Could you both take a look at this for me? @Francisco-Gamino  @JamesWTruher 
Thanks!

### CODE REVIEW

**Comment by daxian-dbw** [comment]
Please fix the test failures in Travis CI.

**Comment by daxian-dbw** [comment]
> test/powershell/Modules/Microsoft.PowerShell.Management/PSDrive.Tests.ps1

The test is about PSDrive, which doesn't belong to Microsoft.PowerShell.Management module. Please move the file to an appropriate folder.
@Francisco-Gamino could you please review this PR?

**Comment by ChrisUbben** [comment]
I talked to @JamesWTruher and he said to put these provider tests in Modules/Microsoft.PowerShell.Management/.

File: test/powershell/Modules/Microsoft.PowerShell.Management/PSDrive.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    AfterEach {
+        Remove-PSDrive -Name $psDriveName -ErrorAction SilentlyContinue
+    }
+
+    It "Create a new PSDrive" {
+        $newDrive = New-PSDrive -Name "NewDifferentPSDrive" -PSProvider FileSystem -Root $psDriveRoot
+        $newDrive.Name | Should Be "NewDifferentPSDrive"
+        $newDrive.Root | Should Be (Convert-Path $psDriveRoot)
+        Remove-PSDrive -Name "NewDifferentPSDrive" -ErrorAction SilentlyContinue
```
Comment (daxian-dbw) [inline test/powershell/Modules/Microsoft.PowerShell.Management/PSDrive.Tests.ps1]: `Remove-PSDrive` should be put in a finally block, so even if the above `Should Be` failed, the test drive can still be cleaned up.

File: test/powershell/Modules/Microsoft.PowerShell.Management/PSDrive.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    It "Read data from a PSDrive" {
+        $driveProp = Get-ItemProperty ${psDriveName}:
+        $driveProp.PSDrive | Should Be $psDriveName
```
Comment (daxian-dbw) [inline test/powershell/Modules/Microsoft.PowerShell.Management/PSDrive.Tests.ps1]: maybe it should be `$driveProp.PSDrive.Name | Should Be $psDriveName`

**Review by daxian-dbw — CHANGES_REQUESTED** [review]

File: test/powershell/Modules/Microsoft.PowerShell.Management/PSDrive.Tests.ps1 (hunk 1 above)
Comment (ChrisUbben) [inline test/powershell/Modules/Microsoft.PowerShell.Management/PSDrive.Tests.ps1]: Fixed. Thanks Dongbo!

File: test/powershell/Modules/Microsoft.PowerShell.Management/PSDrive.Tests.ps1 (hunk 2 above)
Comment (ChrisUbben) [inline test/powershell/Modules/Microsoft.PowerShell.Management/PSDrive.Tests.ps1]: Fixed. Thanks Dongbo!

**Review by daxian-dbw — APPROVED** [review]

**Pull request merged**